internal/controllers/device_data_controller.go

```diff
@@ -215,10 +215,11 @@ func (d *DeviceDataController) getHistory(c *fiber.Ctx, userDevice *grpc.UserDev
 				Query: &types.Query{
 					Bool: &types.BoolQuery{
 						Filter: []types.Query{
-							{Term: map[string]types.TermQuery{"subject": {Value: userDevice.Id}}},
-							{Range: map[string]types.RangeQuery{"data.timestamp": types.DateRangeQuery{Gte: some.String(startDate), Lte: some.String(endDate)}}},
+							{Match: map[string]types.MatchQuery{"subject": {Query: userDevice.Id}}},
```

Comment:
Why did you have to do this?

Reply:
It seems like the term query searches for the exact value where the match query searches for any matches... my suspicion is that keywords are playing in a role in why the two queries turn up different results. 

Seems like the difference lies in how [Elasticsearch](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-term-query.html#:~:text=To%20better%20search%20text%20fields,the%20exact%20term%20you%20provide) applies an analyzer for each respective query type. I found that `Match` returned the expected response for both indexes but `Term` didn't return anything for the vss index

Reply:
Then I think the problem is our mappings. Let me look.